packages/experimental/surface/src/framework/Context.tsx

```diff
@@ -0,0 +1,19 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { FC, PropsWithChildren } from 'react';
+
+import { Plugin } from './Plugin';
+
+export const compose = (contexts: FC<PropsWithChildren>[]) => {
```

Comment:
I know we've discussed -- you've explored -- both tracks here so I'm curious how you landed back with composable contexts?

we have a root plugin context. what additional benefit do we get from each plugin providing a unique context. 
my understanding of contexts is that they just manage a value defined by the provider. what's the advantage here over managing that value (by plugin type) in a root plugin provider?

Reply:
my intuition here is that storing plugin state in a class backed by reactivity that @wittjosiah is doing right now would be simpler